src/qt/walletframe.h

```diff
@@ -17,6 +17,7 @@ class WalletView;
 
 QT_BEGIN_NAMESPACE
 class QStackedWidget;
+class QWidget;
```

Comment:
1f913ac9a00b982c997e70603ba62eebbe4c9d03

Why? `QFrame` extends it so it's available

Reply:
[Developer Notes - Source code organization](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization):
> - Every `.cpp` and `.h` file should `#include` every header file it directly uses classes, functions or other definitions from, even if those headers are already included indirectly through other headers.
>    - _Rationale_: Excluding headers because they are already indirectly included results in compilation failures when those indirect dependencies change. Furthermore, it obscures what the real code dependencies are.

Reply:
Doest that apply for forward declarations?

Reply:
I think it's okay to leave it, though FWIW forward-declaring QWidget should never be necessary. 
It is impossible to derive from `QFrame` if the full definition of `QWidget` (which is its ancestor) isn't available.